third_party/move/move-vm/types/src/loaded_data/runtime_types.rs

```diff
@@ -309,9 +312,8 @@ impl Type {
             TyParam(_) | Bool | U8 | U16 | U32 | U64 | U128 | U256 | Address | Signer => {
                 Self::LEGACY_BASE_MEMORY_SIZE
             },
-            Vector(ty) | Reference(ty) | MutableReference(ty) => {
-                Self::LEGACY_BASE_MEMORY_SIZE + ty.size()
-            },
+            Reference(ty) | MutableReference(ty) => Self::LEGACY_BASE_MEMORY_SIZE + ty.size(),
```

Comment:
ditto: why separated?

Reply:
The reason why Vector needs to be separated from Reference is rather unfortunate. For Ref/MutRef, the variant has `Box<Type>` whereas Vec has variant type `Arc<Type>`. It seems to me that rust wasn't able to deduce that the two references here can be coerced into the same type using AsRef so they need to be in different branch in order for type check to pass.